src/services/comment.service.ts

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { UseFormResetField } from 'react-hook-form';
+import { Action, ActionTypes } from '../reducer/store';
+import { addComment, deleteComment } from '../api/users.api';
+import { Comment, CommentData } from '../types/Comment';
+
+export function commentService(
+  dispatch: (action: Action) => void,
+  commentsList: Comment[] | null,
+) {
+  function addNewComment(
+    dataToSend: CommentData & { postId: number },
+    setShowSpinner: React.Dispatch<React.SetStateAction<boolean>>,
+    resetField: UseFormResetField<CommentData>,
+  ) {
+    addComment(dataToSend)
+      .then((res) => {
+        dispatch({
+          type: ActionTypes.getComments,
+          comments: commentsList ? [...commentsList, res] : [res],
+        });
+
+        resetField('body');
+      })
+      .finally(() => {
```

Comment:
The spinner is only hidden after the addComment operation. If an error occurs during the operation, the spinner will keep loading indefinitely. Consider adding error handling to stop the spinner.